proxy/http/HttpSM.h

```diff
@@ -166,6 +166,52 @@ enum HttpPluginTunnel_t {
 
 class PluginVCCore;
 
+class PendingAction
+{
+public:
+  bool
+  operator==(Action *b)
+  {
+    return b == pending_action;
+  }
+  bool
+  operator!=(Action *b)
+  {
+    return b != pending_action;
+  }
+  PendingAction &
+  operator=(Action *b)
+  {
+    // Don't do anything if the new action is _DONE
+    if (b != ACTION_RESULT_DONE) {
+      if (b != pending_action && pending_action != nullptr) {
+        pending_action->cancel();
+      }
+      pending_action = b;
+    }
+    return *this;
+  }
+  Action *
+  operator->()
+  {
+    return pending_action;
+  }
+  Action *
+  get()
+  {
+    return pending_action;
+  }
+  ~PendingAction()
```

Comment:
Is this destructor really called? It'd be called if HttpSM's destructor was called, but I don't think HttpSM's destructor is called because the allocator declaration don't have `true` to call the destructor automatically and I don't see any places that calls the destructor manually.
```
ClassAllocator<HttpSM> httpSMAllocator("httpSMAllocator");
```

Reply:
Good point.  The pending_action is explicitly assigned to nullptr in the HttpSM::kill_this().  But you are correct.  In the current code, the destructor is not called.

Reply:
No - if this is ever used more normally (e.g. we stop using free lists) then the lack of a destructor will cause some nasty problems. We need to move toward more standard C++, not away from it.

Reply:
I'm not saying we don't need the destructor. My understanding is that this change is to ensure pending_action won't be triggered after HttpSM got unavailable. Manually canceling the action by assigning nullptr is fine. I just thought the current code doesn't fully take the benefit of PendingAction class.

Reply:
Correct, but that's an artifact of the proxy allocators.